embedding/csharp/Tizen.Flutter.Embedding/NUIFlutterView.cs

```diff
@@ -0,0 +1,205 @@
+        /// Starts running the view with the associated engine, creating if not set.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="Engine"/> must not be set again after this call.
+        /// <see cref="Destroy"/> must be called if the view is no longer used.
+        /// </remarks>
+        public bool RunEngine()
+        {
+            if (IsRunning)
+            {
+                TizenLog.Error("The engine is already running.");
+                return false;
+            }
+
+            if (Engine == null)
+            {
+                Engine = new FlutterEngine();
+            }
+
+            if (!Engine.IsValid)
+            {
+                TizenLog.Error("Could not create a Flutter engine.");
+                return false;
+            }
+
+            _viewSize = GetDefaultSize();
+
+            Type baseType = typeof(NativeImageQueue).BaseType.BaseType.BaseType;
+            FieldInfo field = baseType.GetField("swigCPtr", BindingFlags.NonPublic | BindingFlags.Instance);
+            NativeImageQueue nativeImageQueue =
+                new NativeImageQueue((uint)_viewSize.Width, (uint)_viewSize.Height, NativeImageQueue.ColorFormat.RGBA8888);
+            HandleRef nativeImageQueueRef = (HandleRef)field.GetValue(nativeImageQueue);
+            SetImage(nativeImageQueue.GenerateUrl().ToString());
+
+            Type imageViewBaseType = typeof(ImageView).BaseType.BaseType.BaseType.BaseType;
+            FieldInfo imageViewField =
+                imageViewBaseType.GetField("swigCPtr", BindingFlags.NonPublic | BindingFlags.Instance);
+            HandleRef imageViewRef = (HandleRef)imageViewField.GetValue(this);
+
+            var viewProperties = new FlutterDesktopViewProperties
+            {
+                width = _viewSize.Width,
+                height = _viewSize.Height,
+            };
+
+            View = FlutterDesktopViewCreateFromImageView(
+                ref viewProperties, Engine.Engine, imageViewRef.Handle, nativeImageQueueRef.Handle,
+                NUIApplication.GetDefaultWindow().GetNativeId());
+            if (View.IsInvalid)
+            {
+                TizenLog.Error("Could not launch a Flutter view.");
+                return false;
+            }
+
+            RegisterEvent();
+            return true;
+        }
+
+        /// <summary>
+        /// Terminates the running view and the associated engine.
+        /// </summary>
+        public void Destroy()
+        {
+            if (IsRunning)
+            {
+                FlutterDesktopViewDestroy(View);
+                Engine = null;
+                View = new FlutterDesktopView();
+            }
+        }
+
+        /// <summary>
+        /// Returns the size that can be the default.
+        /// </summary>
+        private Size2D GetDefaultSize()
+        {
+            if (Size2D.Width == 0 || Size2D.Height == 0)
+            {
+                View parent = GetParent() as View;
+                if (!parent)
```

Comment:
Can only `View` have a `ImageView` as a child? Are there other `Container` types that can be a parent of `ImageView`?

Reply:
There is Window.Add() case.
In the case of Window.Add(), I used DefaultWindow() because parent cannot be obtained through GetParent().
There are no other types as far as I know.

Reply:
How about the `Layer` class? An instance of it can be obtained through `Window.Instance.GetDefaultLayer()`.

Reply:
I added layer case and change to use GetDefaultLayer().